### PR TITLE
fix: allow multiple populates

### PR DESCRIPTION
**Pull request by Ugzuzg**
It is currently impossible to populate multiple fields at once.

### CODE REVIEW

**Review by eug-vs — APPROVED** [review]

File: packages/mongoose/src/createMongooseController.ts [hunk 1]
```diff
@@ -54,39 +54,90 @@ export const createMongooseController = <T extends Constructor<{}>>(
 		@openapi.prop()
 		path: string;
 
-		@openapi.prop()
-		$limit?: number;
+		@openapi.prop({
+			rawSchemaOptions: {
```
Comment (pierissimo) [inline packages/mongoose/src/createMongooseController.ts:58]: we can remove the `rawSchemaOptions` wrapping

File: packages/mongoose/src/createMongooseController.ts [hunk 2]
```diff
@@ -54,39 +54,90 @@ export const createMongooseController = <T extends Constructor<{}>>(
...
 		@openapi.prop()
 		$skip?: number;
 
 		@openapi.prop()
-		$where?: object;
+		$limit?: number;
+
+		@openapi.prop({
+			type: null,
+			rawSchemaOptions: {
```
Comment (pierissimo) [inline packages/mongoose/src/createMongooseController.ts:76]: as above

File: packages/mongoose/src/createMongooseController.ts [hunk 3]
```diff
@@ -54,39 +54,90 @@ export const createMongooseController = <T extends Constructor<{}>>(
...
+						items: { type: 'string' }
+					}
+				]
+			}
+		})
+		$select?: string | string[];
 
 		@openapi.prop({
+			rawSchemaOptions: {
+				properties: {}
```
Comment (pierissimo) [inline packages/mongoose/src/createMongooseController.ts:92]: is this needed?

File: packages/mongoose/src/createMongooseController.ts [hunk 4]
```diff
@@ -54,39 +54,90 @@ export const createMongooseController = <T extends Constructor<{}>>(
...
-		@openapi.prop()
-		$populate?: PopulateQueryParameter;
+		$populate?: PopulateQueryParameter | string | (PopulateQueryParameter | string)[];
 	}
 
 	@openapi.definition({ title: 'QueryParameters' })
 	class QueryParameters {
-		@openapi.prop({ type: PopulateQueryParameter })
-		$populate?: PopulateQueryParameter;
+		// davinci doesn't support class definitions in oneOf yet
```
Comment (pierissimo) [inline packages/mongoose/src/createMongooseController.ts:119]: let's setup a call to discuss this 👍

**Pull request closed without merging**